controllers/disruption_schedule_controller.go

```diff
@@ -0,0 +1,221 @@
+	mostRecentScheduleTime := r.getMostRecentScheduleTime(disruptions) // find the last run so we can update the status
+	if mostRecentScheduleTime != nil {
+		instance.Status.LastScheduleTime = &metav1.Time{Time: *mostRecentScheduleTime}
+	} else {
+		instance.Status.LastScheduleTime = nil
+	}
+
+	return r.Client.Status().Update(ctx, instance)
+}
+
+// getMostRecentScheduleTime returns the pointer to the most recent scheduled time among a list of disruptions
+func (r *DisruptionScheduleReconciler) getMostRecentScheduleTime(disruptions *chaosv1beta1.DisruptionList) *time.Time {
+	var mostRecentScheduleTime *time.Time
+
+	for _, disruption := range disruptions.Items {
+		scheduledTimeForDisruption, err := r.getScheduledTimeForDisruption(&disruption)
+		if err != nil {
+			r.log.Errorw("unable to parse schedule time for child disruption", "err", err, "disruption", &disruption)
+			continue
+		}
+
+		if scheduledTimeForDisruption != nil {
+			if mostRecentScheduleTime == nil {
+				mostRecentScheduleTime = scheduledTimeForDisruption
+			} else if mostRecentScheduleTime.Before(*scheduledTimeForDisruption) {
+				mostRecentScheduleTime = scheduledTimeForDisruption
+			}
+		}
+	}
+
+	return mostRecentScheduleTime
+}
+
+// getScheduledTimeForDisruption returns the scheduled time for a particular disruption
+func (r *DisruptionScheduleReconciler) getScheduledTimeForDisruption(disruption *chaosv1beta1.Disruption) (*time.Time, error) {
+	timeRaw := disruption.Annotations[ScheduledAtAnnotation]
+	if len(timeRaw) == 0 {
+		return nil, nil
+	}
+
+	timeParsed, err := time.Parse(time.RFC3339, timeRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	return &timeParsed, nil
+}
+
+func (r *DisruptionScheduleReconciler) checkTargetResourceExists(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) (bool, error) {
+	var targetObj client.Object
+
+	switch instance.Spec.TargetResource.Kind {
+	case "Deployment":
+		targetObj = &appsv1.Deployment{}
+	case "StatefulSet":
+		targetObj = &appsv1.StatefulSet{}
+	}
+
+	err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      instance.Spec.TargetResource.Name,
+		Namespace: instance.Namespace,
+	}, targetObj)
+
+	return errors.IsNotFound(err), err
+}
+
+func (r *DisruptionScheduleReconciler) updateTargetResourcePreviouslyMissing(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) (bool, bool, error) {
+	disruptionScheduleDeleted := false
+	targetResourceNotFound, err := r.checkTargetResourceExists(ctx, instance)
+
+	if targetResourceNotFound {
+		r.log.Warnw("target does not exist, this schedule will be deleted if that continues", "error", err)
+
+		if instance.Status.TargetResourcePreviouslyMissing == nil {
+			r.log.Warnw("target is missing for the first time, updating status", "targetPreviouslyMissing", instance.Status.TargetResourcePreviouslyMissing)
```

Comment:
```suggestion
			r.log.Warn("target is missing for the first time, updating status")
```

Why add this annotation if `targetPreviouslyMissing` will always be `nil` here?